operatorcert/pyxis.py

```diff
@@ -1,19 +1,71 @@
 import logging
 import os
 from typing import Any, Dict
+from urllib.parse import urljoin
 
 import requests
 
 LOGGER = logging.getLogger("operator-cert")
 
 
-def post_with_api_key(url: str, body: Dict[str, Any]) -> Dict[str, Any]:
+def is_internal() -> bool:
+    """
+    Check if provided configuration points to internal vs external Pyxis instance
+
+    Returns:
+        bool: Is internal Pyxis instance
+    """
+    cert = os.environ.get("PYXIS_CERT_PATH")
+    key = os.environ.get("PYXIS_KEY_PATH")
+    return cert and key
+
+
+def _get_session() -> requests.Session:
+    """
+    Create a Pyxis http session with auth based on env variables.
+
+    Auth is set to use either API key or certificate + key.
+
+    Raises:
+        Exception: Exception is raised when auth ENV variables are missing.
+
+    Returns:
+        requests.Session: Pyxis session
+    """
     api_key = os.environ.get("PYXIS_API_KEY")
-    if not api_key:
-        raise Exception("Pyxis API key is missing. Define PYXIS_API_KEY env variable")
+    cert = os.environ.get("PYXIS_CERT_PATH")
+    key = os.environ.get("PYXIS_KEY_PATH")
+
+    # API key or cert + key need to be provided using env variable
+    if not api_key and (not cert or not key):
+        raise Exception(
+            "No auth details provided for Pyxis. "
+            "Either define PYXIS_API_KEY or PYXIS_CERT_PATH + PYXIS_KEY_PATH"
+        )
+
+    session = requests.Session()
+    if api_key:
+        session.headers.update({"X-API-KEY": api_key})
+    else:
+        session.cert = (cert, key)
+    return session
+
+
+def post(url: str, body: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
FYI I just merged a change with additional post and get with certs that are done differently in https://github.com/redhat-openshift-ecosystem/operator-pipelines-images/pull/12, but I think your way is better, so I'm happy to rip out all of that and update the reserve-operator-name script to use this one instead once you have it merged. There'll probably be conflicts for you though.